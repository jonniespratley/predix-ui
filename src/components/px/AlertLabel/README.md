The `<AlertLabel/>` component labels are used to draw attention to a particular piece of information on the screen.


## Usage

```react
---
const {AlertLabel} = PxReact;
<AlertLabel
  type='info'
  label='Info'/>
```

```react
---
const {AlertLabel} = PxReact;
<AlertLabel 
  type='warning' 
  label='warning'/>
```

```react
---
const {AlertLabel} = PxReact;
<AlertLabel 
  type='healthy' 
  label='healthy'/>
```

```react
---
const {AlertLabel} = PxReact;
<AlertLabel 
  type='error'
  label='error'/>
```

```react
---
const {AlertLabel} = PxReact;
<AlertLabel 
  type='important'
  label='important'/>
```

```react
---
const {AlertLabel} = PxReact;
<AlertLabel 
  type='unknown'
  label='unknown'/>
```


## Styling
The following custom properties are available for styling

```table
rows:
  - Custom property: --px-alert-label-border-color
    Description: Border or outline color for all labels
  
  - Custom property: --px-alert-label-background-color--alert
    Description: Background color for 'alert' type labels
  
  - Custom property: --px-alert-label-text-color--alert
    Description: Text color for 'alert' type labels
  
  - Custom property: --px-alert-label-background-color--warning
    Description: Background color for 'warning' type labels
  
  - Custom property: --px-alert-label-text-color--warning
    Description: Text color for 'warning' type labels
  
  - Custom property: --px-alert-label-background-color--error
    Description: Background color for 'error' type labels
  
  - Custom property: --px-alert-label-text-color--error
    Description: Text color for 'error' type labels
  
  - Custom property: --px-alert-label-background-color--information
    Description: Background color for 'information' type labels
  
  - Custom property: --px-alert-label-text-color--information
    Description: Text color for 'information' type labels
  
  - Custom property: --px-alert-label-background-color--healthy
    Description: Background color for 'healthy' type labels
  
  - Custom property: --px-alert-label-text-color--healthy
    Description: Text color for 'healthy' type labels
  
  - Custom property: --px-alert-label-background-color--unknown
    Description: Background color for 'unknown' type labels
  
  - Custom property: --px-alert-label-text-color--unknown
    Description: Text color for 'unknown' type labels
```

