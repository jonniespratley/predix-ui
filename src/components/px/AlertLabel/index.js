import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const AlertLabel = styled.div`
  display         : inline-block;
  position        : relative;
  padding         : 5px;
  text-transform  : uppercase;
  text-align      : center;
  font-size       : 0.8rem;
  line-height     : 0.8rem;
  background-color: var(--px-alert-label-background-color--unknown, gray);
  color           : var(--px-alert-label-text-color--unknown, white);
  border          : 1px solid var(--px-alert-label-border-color, transparent);

  ${props => props.type === 'info' && css`
    background-color: var(--px-alert-label-background-color--information, blue);
    color           : var(--px-alert-label-text-color--information, white);
  `}
  ${props => props.type === 'error' && css`
    background-color: var(--px-alert-label-background-color--error, yellow);
    color           : var(--px-alert-label-text-color--error, black);
  `}
  ${props => props.type === 'unknown' && css`
    background-color: var(--px-alert-label-background-color--unknown, gray);
    color           : var(--px-alert-label-text-color--unknown, white);
  `}
  ${props => props.type === 'important' && css`
    background-color: var(--px-alert-label-background-color--important, red);
    color           : var(--px-alert-label-text-color--important, white);
  `}
  ${props => props.type === 'warning' && css`
    background-color: var(--px-alert-label-background-color--warning, orange);
    color           : var(--px-alert-label-text-color--warning, white);
  `}
  ${props => props.type === 'healthy' && css`
    background-color: var(--px-alert-label-background-color--healthy, green);
    color           : var(--px-alert-label-text-color--healthy, white);
  `}
  ${props => props.type === 'success' && css`
    background-color: var(--px-alert-label-background-color--healthy, green);
    color           : var(--px-alert-label-text-color--healthy, white);
  `}
`;

const Component = ({
  label, type
}) => (
  <AlertLabel type={type}>
    <span className="label__text">{label}</span>
  </AlertLabel>
);

Component.defaultProps = {
  label: null,
  type: null
};

Component.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string
};

Component.displayName = 'AlertLabel';

export default Component;
