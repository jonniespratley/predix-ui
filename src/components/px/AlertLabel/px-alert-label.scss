//@import "px-normalize-design/_generic.normalize.scss";
//@import "px-box-sizing-design/_generic.box-sizing.scss";
/**
 * px-alert-label styles
 */
.px-alert-label {
  // Component

  .alertlabel {
    display         : inline-block;
    position        : relative;
    padding         : calculateRem(5px);
    text-transform  : uppercase;
    text-align      : center;
    font-size       : 0.8rem;
    line-height     : 0.8rem;
    background-color: var(--px-alert-label-background-color--important, red);
    color           : var(--px-alert-label-text-color--important, white);
    border          : 1px solid var(--px-alert-label-border-color, transparent);
  }

  .svg-canvas {
    width : 2.2rem;
    height: 2.2rem;

    #circle,
    #polygon {
      stroke-width: 1px;
      stroke      : var(--px-alert-label-border-color, transparent);
    }
  }

  .badge {
    padding: 0;

    .label {
      display        : flex;
      position       : absolute;
      top            : 0;
      right          : 0;
      bottom         : 0;
      left           : 0;
      width          : 100%;
      align-items    : center;
      justify-content: center;
    }
  }

  .important {
    background-color: var(--px-alert-label-background-color--important, red);
    color           : var(--px-alert-label-text-color--important, white);

    &.badge {
      background: none;
      border    : none;
      fill      : var(--px-alert-label-background-color--important, red);

      .label {
        top: calculateRem(8px);
      }
    }
  }

  .warning {
    background-color: var(--px-alert-label-background-color--warning, orange);
    color           : var(--px-alert-label-text-color--warning, white);

    &.badge {
      background: none;
      border    : none;
      fill      : var(--px-alert-label-background-color--warning, orange);
    }
  }

  .error {
    background-color: var(--px-alert-label-background-color--error, yellow);
    color           : var(--px-alert-label-text-color--error, black);

    &.badge {
      background: none;
      border    : none;
      fill      : var(--px-alert-label-background-color--error, yellow);
    }
  }

  .info,
  .information {
    background-color: var(--px-alert-label-background-color--information, blue);
    color           : var(--px-alert-label-text-color--information, white);

    &.badge {
      background: none;
      border    : none;
      fill      : var(--px-alert-label-background-color--information, blue);

      .label {
        top: calculateRem(2px);
      }
    }
  }

  .healthy {
    background-color: var(--px-alert-label-background-color--healthy, green);
    color           : var(--px-alert-label-text-color--healthy, white);

    &.badge {
      background: none;
      border    : none;
      fill      : var(--px-alert-label-background-color--healthy, green);
    }
  }

  .unknown {
    background-color: var(--px-alert-label-background-color--unknown, gray);
    color           : var(--px-alert-label-text-color--unknown, white);

    &.badge {
      background: none;
      border    : none;
      fill      : var(--px-alert-label-background-color--unknown, gray);

      .label {
        top: calculateRem(-1px);
      }
    }
  }
}
