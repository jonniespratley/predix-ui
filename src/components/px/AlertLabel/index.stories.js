import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, select } from '@storybook/addon-knobs/react';
import { withReadme } from 'storybook-readme';

import AlertLabel from './';
import README from './README.md';

const typeOptions = {
  info: 'Info',
  error: 'Error',
  important: 'Important',
  healthy: 'Healthy',
  warning: 'Warning',
  success: 'Success'
};

storiesOf('Components / Alert Label', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(README))
  .add('default', () => (
    <AlertLabel
      label={text('label', 'Info')}
      badge={text('badge', '1')}
      type={select('type', typeOptions, 'info', 'alertType')}
    />
  ))
  .add('error', () => (
    <AlertLabel
      label="Error"
      type="error"
    />
  ))
  .add('important', () => (
    <AlertLabel
      label="Important"
      type="important"
    />
  ))
  .add('healthy', () => (
    <AlertLabel
      label="Healthy"
      type="healthy"
    />
  ))
  .add('warning', () => (
    <AlertLabel
      label="Warning"
      type="warning"
    />
  ))
  .add('unknown', () => (
    <AlertLabel
      type="unknown"
      label="unknown"
    />
  ));

