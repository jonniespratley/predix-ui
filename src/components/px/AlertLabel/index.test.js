//import { expect } from 'chai';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AlertLabel from './';
import 'jest-styled-components'

const types = [
	'healthy',
	'important',
	'warning',
	'error',
	'info',
	'success',
	'unknown'
];
describe('AlertLabel', () => {
  test('should render', () => {
    const wrapper = shallow(<AlertLabel/>);
    expect(wrapper);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<AlertLabel type='info' />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('background-color', 'var(--px-alert-label-background-color--information,blue)');
    expect(tree).toHaveStyleRule('color', 'var(--px-alert-label-text-color--information,white)');
  });

  test('should render correctly', () => {
    const wrapper = shallow(
      <AlertLabel label='Test'/>
    );
    expect(wrapper.find('.label'));
  });
  types.forEach(type => {
    test(`should render type = ${type}`, () => {
      const wrapper = shallow(<AlertLabel type={type} label={type}/>);
      expect(wrapper.props().type).toEqual(type);
      const tree = renderer.create(<AlertLabel type={type} label={type} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

});
