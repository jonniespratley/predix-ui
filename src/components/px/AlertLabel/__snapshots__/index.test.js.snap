// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertLabel matches snapshot 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  padding: 5px;
  text-transform: uppercase;
  text-align: center;
  font-size: 0.8rem;
  line-height: 0.8rem;
  background-color: var(--px-alert-label-background-color--unknown,gray);
  color: var(--px-alert-label-text-color--unknown,white);
  border: 1px solid var(--px-alert-label-border-color,transparent);
  background-color: var(--px-alert-label-background-color--information,blue);
  color: var(--px-alert-label-text-color--information,white);
}

<div
  className="c0"
  type="info"
>
  <span
    className="alertlabel info"
  >
    <div
      className="label"
    >
      <span
        className="label__text"
      />
    </div>
  </span>
</div>
`;
